x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackClientPlugin.java

```diff
@@ -534,6 +535,11 @@ public List<NamedWriteableRegistry.Entry> getNamedWriteables() {
                     TransformField.TIME.getPreferredName(),
                     TimeRetentionPolicyConfig::new
                 ),
+                new NamedWriteableRegistry.Entry(
```

Comment:
`TransformNamedXContentProvider` is also missing the `NullRetentionPolicyConfig` named writable

Reply:
@davidkyle it isn't required. This thing is never written to XContent. But I can add it for completeness.

Reply:
Please do I think it is worth having to avoid a similar problem in future